### PR TITLE
Adds HLSThreshold Operation

### PR DESCRIPTION
**Pull request by JLLeitschuh**
Allows users to perform the HLS Threhold operation on an image

Related #86

### CODE REVIEW

File: src/main/java/edu/wpi/grip/core/operations/composite/HLSThresholdOperation.java [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+import static org.bytedeco.javacpp.opencv_imgproc.COLOR_BGR2HLS;
+import static org.bytedeco.javacpp.opencv_imgproc.cvtColor;
+
+/**
+ * An {@link edu.wpi.grip.core.Operation} that converts a color image into a binary image based on the HLS threshold ranges for each channel
+ */
+public class HLSThresholdOperation implements CVOperation {
+    private final SocketHint<opencv_core.Mat> inputHint = new SocketHint<opencv_core.Mat>("Input", opencv_core.Mat.class, opencv_core.Mat::new);
+    private final SocketHint<List> hueHint = new SocketHint<List>("Hue", List.class,
+            () -> Arrays.asList(0.0, 255.0), SocketHint.View.RANGE, new List[]{Arrays.asList(0.0, 180.0)});
```
Comment (ThomasJClark) [inline src/main/java/edu/wpi/grip/core/operations/composite/HLSThresholdOperation.java]: The default value should be (0.0, 180.0)

File: src/main/java/edu/wpi/grip/core/operations/composite/HLSThresholdOperation.java [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    @Override
+    public String getDescription() {
+        return "Segment an image based on hue, luminance and saturation ranges";
+    }
+
+    @Override
+    public InputSocket<?>[] createInputSockets(EventBus eventBus) {
+        return new InputSocket<?>[]{
+                new InputSocket<>(eventBus, inputHint),
+                new InputSocket<>(eventBus, hueHint),
```
Comment (ThomasJClark) [inline src/main/java/edu/wpi/grip/core/operations/composite/HLSThresholdOperation.java]: It might make more sense to make them in HSL order

Comment (JLLeitschuh) [inline src/main/java/edu/wpi/grip/core/operations/composite/HLSThresholdOperation.java]: I was following the CV standards. Is this an idea or something you expect to be changed?
I don't care myself.

Comment (ThomasJClark) [inline src/main/java/edu/wpi/grip/core/operations/composite/HLSThresholdOperation.java]: https://en.wikipedia.org/wiki/HSL_and_HSV

Comment (ThomasJClark) [inline src/main/java/edu/wpi/grip/core/operations/composite/HLSThresholdOperation.java]: I don't think there's a computer vision standard for HLS

File: src/main/java/edu/wpi/grip/core/operations/composite/HSLThresholdOperation.java [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+
+import java.util.Arrays;
+import java.util.List;
+
+import static org.bytedeco.javacpp.opencv_core.inRange;
+import static org.bytedeco.javacpp.opencv_imgproc.COLOR_BGR2HLS;
+import static org.bytedeco.javacpp.opencv_imgproc.cvtColor;
+
+/**
+ * An {@link edu.wpi.grip.core.Operation} that converts a color image into a binary image based on the HSL threshold ranges for each channel
```
Comment (ThomasJClark) [inline src/main/java/edu/wpi/grip/core/operations/composite/HSLThresholdOperation.java]: Not "for each channel" - the channels are converted into HSL

File: src/main/java/edu/wpi/grip/core/operations/composite/HSVThresholdOperation.java [hunk 4]
```diff
@@ -21,7 +21,7 @@
 
     private final SocketHint<opencv_core.Mat> inputHint = new SocketHint<opencv_core.Mat>("Input", opencv_core.Mat.class, opencv_core.Mat::new);
     private final SocketHint<List> hueHint = new SocketHint<List>("Hue", List.class,
-            () -> Arrays.asList(0.0, 255.0), SocketHint.View.RANGE, new List[]{Arrays.asList(0.0, 180.0)});
+            () -> Arrays.asList(0.0, 180.0), SocketHint.View.RANGE, new List[]{Arrays.asList(0.0, 180.0)});
```
Comment (JLLeitschuh) [inline src/main/java/edu/wpi/grip/core/operations/composite/HSVThresholdOperation.java]: Your comment made me remember that I didn't fix this. This is in a separate commit. Hope you don't mind

File: src/main/java/edu/wpi/grip/core/operations/composite/HSLThresholdOperation.java [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+                channel1.get(0).doubleValue(),
+                channel3.get(0).doubleValue(),
+                channel2.get(0).doubleValue(), 0));
+
+        final opencv_core.Mat high = new opencv_core.Mat(input.size(), input.type(), new opencv_core.Scalar(
+                channel1.get(1).doubleValue(),
+                channel3.get(1).doubleValue(),
+                channel2.get(1).doubleValue(), 0));
+
+        final opencv_core.Mat hls = new opencv_core.Mat();
```
Comment (ThomasJClark) [inline src/main/java/edu/wpi/grip/core/operations/composite/HSLThresholdOperation.java]: This is probably just something intellij automatically messed up when you copied and pasted, but open_core.* is usually statically imported.

File: src/main/java/edu/wpi/grip/core/operations/composite/HSLThresholdOperation.java [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+    private final SocketHint<opencv_core.Mat> outputHint = new SocketHint<opencv_core.Mat>("output", opencv_core.Mat.class, opencv_core.Mat::new);
+
+    @Override
+    public String getName() {
+        return "HSL Threshold";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Segment an image based on hue, saturation and luminance ranges";
```
Comment (ThomasJClark) [inline src/main/java/edu/wpi/grip/core/operations/composite/HSLThresholdOperation.java]: *hue, saturation, and luminance

Comment (JLLeitschuh) [inline src/main/java/edu/wpi/grip/core/operations/composite/HSLThresholdOperation.java]: Dam oxford comma

**Comment by JLLeitschuh** [comment]
Dam oxford comma

**Pull request merged**